Backends/CLX/medium.lisp

```diff
@@ -361,6 +361,21 @@ translated, so they begin at different position than [0,0])."))
 
 ;;; XXX: both PM and MM pixmaps should be freed with (xlib:free-pixmap pixmap)
 ;;; when not used. We do not do that right now.
+(defun put-image-recursively (pixmap pixmap-context pixmap-image width height x0 y0)
+  (cond
+    ((and (< width 2048) (< height 2048))
+     (xlib:put-image pixmap pixmap-context pixmap-image :src-x x0 :src-y y0 :x x0 :y y0
+                     :width width :height height))
+    ((> width height)
+     (put-image-recursively pixmap pixmap-context pixmap-image (ceiling width 2) height x0 y0)
```

Comment:
You could do the recursion using a local `labels` function to avoid passing the non-changing arguments over and over. That would also be more efficient in some implementations.

Reply:
All right, but how large a pixmap are we expecting here? Seeing as this software rendering business will be obsolete in [optimistically short amount of time], I personally think optimizing for cycles is maybe best done at another place.

Reply:
True. I see the main benefit in not repeating the function name and unchanged arguments.

Reply:
Agreed, I put the change in.